src/theory/builtin/kinds

```diff
@@ -296,7 +296,8 @@ operator EQUAL 2 "equality (two parameters only, sorts must match)"
 operator DISTINCT 2: "disequality (N-ary, sorts must match)"
 variable VARIABLE "a variable (not permitted in bindings)"
 variable BOUND_VARIABLE "a bound variable (permitted in bindings and the associated lambda and quantifier bodies only)"
-variable SKOLEM "a Skolem variable (internal only)"
+variable SKOLEM "a Skolem variable"
+variable DUMMY_SKOLEM "a Skolem variable (internal only)"
```

Comment:
Why not add `DUMMY_SKOLEM` as an `InternalSkolemId`?

Reply:
Skolems constructed via `InternalSkolemId` are constructed based on a `cacheVal` and are such that the `SkolemManager` caches them, they have `SKOLEM` kind but skolem id `INTERNAL`.  This makes it simple that the set of skolems cached in `SkolemManager` are exactly those with kind `SKOLEM`.

Dummy skolems are even less restricted: they are just simply fresh variables constructed without caching.  Making them have a different kind simplifies a lot of the logic.  

In the longer term, we should get rid of all calls to `mkDummySkolem`.